File: check_update.py
#!/usr/bin/env python3
"""德仁口腔AI助手 - 自动更新脚本
从GitHub API拉取最新文件，不依赖git命令，纯HTTP更新。
"""
import json, base64, sys, os
from urllib.request import Request, urlopen
from urllib.error import URLError

OWNER = "mahousen"
REPO = "dr-ai-v2"
API = f"https://api.github.com/repos/{OWNER}/{REPO}/contents"
TIMEOUT = 15

def fetch_file(path):
    req = Request(f"{API}/{path}")
    with urlopen(req, timeout=TIMEOUT) as resp:
        data = json.loads(resp.read())
        return base64.b64decode(data["content"])

def get_remote_version():
    content = fetch_file("version.txt")
    return content.decode("utf-8").strip()

def get_local_version():
    try:
        with open("version.txt", "r", encoding="utf-8") as f:
            return f.read().strip()
    except:
        return ""

def download_and_save(fname):
    content = fetch_file(fname)
    with open(fname, "wb") as f:
        f.write(content)

def main():
    print("检查更新...", end=" ", flush=True)
    try:
        remote = get_remote_version()
        local = get_local_version()
    except URLError:
        print("网络不可达，跳过更新")
        return
    except Exception as e:
        print(f"检查失败: {e}")
        return

    if local == remote:
        print(f"已是最新 ({local})")
        return

    print(f"发现新版本: {remote} (当前: {local or '无'})")
    print("正在更新...")

    files = ["index.html", "server.js", "version.txt", "check_update.py"]
    for fname in files:
        try:
            download_and_save(fname)
            print(f"  OK  {fname}")
        except Exception as e:
            print(f"  ERR {fname}: {e}")

    print("更新完成！")

if __name__ == "__main__":
    main()

File: check_update.js
// 德仁口腔AI助手 - Node.js 版自动更新（无需 Python）
const https = require('https');
const fs = require('fs');

const OWNER = 'mahousen';
const REPO = 'dr-ai-v2';

function fetchFile(name) {
  return new Promise((resolve, reject) => {
    const options = {
      hostname: 'api.github.com',
      path: `/repos/${OWNER}/${REPO}/contents/${name}`,
      headers: { 'User-Agent': 'dr-ai-v2-updater' }
    };
    https.get(options, (res) => {
      let data = '';
      res.on('data', d => data += d);
      res.on('end', () => {
        try {
          const json = JSON.parse(data);
          if (json.content) {
            resolve(Buffer.from(json.content, 'base64').toString('utf-8'));
          } else {
            reject(new Error(json.message || 'Unknown error'));
          }
        } catch (e) { reject(e); }
      });
    }).on('error', reject);
  });
}

async function main() {
  try {
    const remote = (await fetchFile('version.txt')).trim();
    let local = '';
    try { local = fs.readFileSync('version.txt', 'utf-8').trim(); } catch (_) {}

    if (local === remote) {
      console.log(`已是最新版本 (${local})`);
      return;
    }

    console.log(`发现新版本: ${remote} (当前: ${local || '无'})`);
    console.log('正在更新...');

    const files = ['index.html', 'server.js', 'version.txt', 'check_update.js', 'check_update.py', 'package.json'];
    for (const f of files) {
      try {
        const content = await fetchFile(f);
        fs.writeFileSync(f, content, 'utf-8');
        console.log(`  OK  ${f}`);
      } catch (e) {
        console.log(`  ERR ${f}: ${e.message}`);
      }
    }
    console.log('更新完成！');
  } catch (e) {
    console.log('检查更新失败:', e.message);
  }
}

main();
